lib/client/api.go

```diff
@@ -3016,11 +3018,6 @@ func (tc *TeleportClient) Login(ctx context.Context) (*Key, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	// Set private key policy from ping response if not already set.
-	if tc.PrivateKeyPolicy == "" {
-		tc.PrivateKeyPolicy = pr.Auth.PrivateKeyPolicy
-	}
-
```

Comment:
I also think this might be useful, in case the profile saved policy is out of sync with cluster auth preference?

I tried this scenario:
* disable hardware key policy in roles/cluster auth preference
* login with tsh
* modify cluster auth preference to be "hardware_key_touch"
* run a tsh command

Interestingly I got this:
```
$ tsh ls                                                                    (base) 
Unmet private key policy "hardware_key_touch"
Relogging in with YubiKey generated private key.

			WARNING
			Detected potentially incompatible client and server versions.
			Minimum client version supported by the server is 12.0.0 but you are using 12.0.0-alpha.1.
			Please upgrade tsh to 12.0.0 or newer or use the --skip-version-check flag to bypass this check.
			Future versions of tsh will fail when incompatible versions are detected.
			Enter password for Teleport user martian:
Tap any security key
Tap your YubiKey
```

will look into this more on monday.

Reply:
This is still being set from the Ping response, but it's being done directly from `TeleportClient.Ping` like the other fields - https://github.com/gravitational/teleport/blob/cc08ffcf1428490ea3eb323dc11b12fbfdb66887/lib/client/api.go#L3503